ffi/diplomat/wasm/wasm-demo/src/ts/date-time.ts

```diff
@@ -89,25 +119,53 @@ export function setup(dataProvider: ICU4XDataProvider): void {
     const otherLocaleInput = document.getElementById('dtf-locale-other-input') as HTMLInputElement | null;
     otherLocaleInput?.addEventListener('input', () => {
         const otherLocaleBtn = document.getElementById('dtf-locale-other') as HTMLInputElement | null;
-        if (otherLocaleBtn?.checked) {
+        if (otherLocaleBtn != null) {
+            otherLocaleBtn.checked = true;
             dateTimeDemo.setLocale(otherLocaleInput.value);
         }
     });
 
-    for (let btn of document.querySelectorAll<HTMLInputElement | null>('input[name="dtf-locale"]')) {
-        if (btn?.value !== 'other') {
-            btn.addEventListener('click', () => dateTimeDemo.setLocale(btn.value));
+    for (let input of document.querySelectorAll<HTMLInputElement | null>('input[name="dtf-locale"]')) {
+        if (input?.value !== 'other') {
+            input.addEventListener('input', () => {
+                dateTimeDemo.setLocale(input.value)
+            });
         }
     }
+    for (let selector of document.querySelectorAll<HTMLSelectElement | null>('select[name="dtf-calendar"]')) {
+        // <select> doesn't have oninput
+        selector?.addEventListener('change', () => {
+            dateTimeDemo.setCalendar(selector.value)
+        });
+    }
 
-    for (let btn of document.querySelectorAll<HTMLInputElement | null>('input[name="dtf-date-length"]')) {
-        btn?.addEventListener('click', () => dateTimeDemo.setDateLength(btn.value));
+    for (let input of document.querySelectorAll<HTMLInputElement | null>('input[name="dtf-date-length"]')) {
+        input?.addEventListener('input', () => {
+            dateTimeDemo.setDateLength(input.value)
+        });
     }
 
-    for (let btn of document.querySelectorAll<HTMLInputElement | null>('input[name="dtf-time-length"]')) {
-        btn?.addEventListener('click', () => dateTimeDemo.setTimeLength(btn.value));
+    for (let input of document.querySelectorAll<HTMLInputElement | null>('input[name="dtf-time-length"]')) {
+        input?.addEventListener('input', () => {
+            dateTimeDemo.setTimeLength(input.value)
+        });
     }
 
     const inputDateTime = document.getElementById('dtf-input') as HTMLInputElement | null;
-    inputDateTime?.addEventListener('input', () => dateTimeDemo.setDateTime(inputDateTime.value));
-}
\ No newline at end of file
+    inputDateTime?.addEventListener('input', () => {
+        dateTimeDemo.setDateTime(inputDateTime.value)
+    });
+    
+    // Annoyingly `toISOString()` gets us the format we need, but it converts to UTC first
```

Comment:
Thought: This solution doesn't appear to be any worse than other solutions. Here is how you would do it in Temporal:

https://stackoverflow.com/a/73298208/1407170

Reply:
Yeah the input element's `valueAsDate` probably should interop with Temporal, eventually.

(Perhaps valueAsTemporalDate or valueAsDateTime)